httprouter/jsonrpcapi/message.go

```diff
@@ -0,0 +1,62 @@
+package jsonrpcapi
+
+import (
+	"encoding/json"
+	"time"
+
+	"go.vocdoni.io/dvote/crypto"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/log"
+)
+
+// MessageAPI inteface defines the methods that the SignedJRPC custom message type must contain.
+type MessageAPI interface {
+	GetID() string
+	SetID(string)
+	SetTimestamp(int32)
+	SetError(string)
+	GetMethod() string
+}
+
+// RequestMessage is the envelope for the SignedJRPC message request
+type RequestMessage struct {
+	MessageAPI json.RawMessage `json:"request"`
+
+	ID        string   `json:"id"`
+	Signature HexBytes `json:"signature"`
+}
+
+// ResponseMessage is the envelope for the SignedJRPC message response
+type ResponseMessage struct {
+	MessageAPI json.RawMessage `json:"response"`
+
+	ID        string   `json:"id"`
+	Signature HexBytes `json:"signature"`
+}
+
+// BuildReply builds a response message (ID, Timestamp and Signature) and marshals it using JSON.
+// The output message is ready to send as reply to the client using reqMsg.Context.Send(data)
+func BuildReply(signer *ethereum.SignKeys, msg MessageAPI, requestID string) ([]byte, error) {
+	var err error
+	respRequest := &ResponseMessage{ID: requestID}
+	msg.SetID(requestID)
+	msg.SetTimestamp(int32(time.Now().Unix()))
```

Comment:
forcing int32 for timestamps will make our software break by 2038, which is not very far :) why not just use int64, as that's what time.Now().Unix() already returns?

Reply:
This was done this way to avoid precision problems on the javascript API side.

Reply:
Then use integers stored in strings - it's just two more characters, and you entirely sidestep the issue of float precision in json :) In Go, they are natively supported like:

```
type T struct {
    IntAsString int64 `json:",string"`
}
```

Reply:
I should mention - the decoder side (e.g. JS) also needs to take care to decode the integer string as a non-float, such as using a real int64 or a "big int". But I imagine that's doable.

Reply:
Sure, but we are not gonna do breaking changes now. I'll open an issue to change this in the future.

Reply:
https://github.com/vocdoni/vocdoni-node/issues/345